test/opt/fold_test.cpp

```diff
@@ -6941,6 +6941,136 @@ ::testing::Values(
                                  1, false)
 ));
 
+// clang-format on
+using ImageOperandsBitmaskFoldingTest =
+    ::testing::TestWithParam<InstructionFoldingCase<bool>>;
+
+TEST_P(ImageOperandsBitmaskFoldingTest, Case) {
```

Comment:
Why did you have to rewrite this?  I think you could have reused `MatchingInstructionWithNoResultFoldingTest`.  None of your tests test what happens to uses of the result id.  They all want to fold the instruction before the return. 
 Even if the instruction before the return has a result id, it should still get folded by `MatchingInstructionWithNoResultFoldingTest`.  It might just need to be renamed.

Reply:
Ah, yes. I hadn't seen this `MatchingInstructionWithNoResultFoldingTest`.  Done. I reused the same name.